Sources/PostgresNIO/Connection/PostgresRequest.swift

```diff
@@ -5,7 +5,7 @@ extension PostgresConnection: PostgresClient {
         request.log(to: self.logger)
         let promise = self.channel.eventLoop.makePromise(of: Void.self)
         let request = PostgresRequestContext(delegate: request, promise: promise)
-        self.channel.write(request).cascadeFailure(to: promise)
```

Comment:
hmm, I don't think this can _fix_ a promise leak, right? If anything this will make it more likely that the promise is never fulfilled. Also, this is missing a test I think?